src/models/tokenSwap.ts

```diff
@@ -65,472 +64,283 @@ export const TokenSwapLayoutV1: typeof BufferLayout.Structure = BufferLayout.str
-    { pubkey: tokenAccountB, isSigner: false, isWritable: false },
-    { pubkey: tokenPool, isSigner: false, isWritable: true },
-    { pubkey: feeAccount, isSigner: false, isWritable: false },
-    { pubkey: destinationAccount, isSigner: false, isWritable: true },
-    { pubkey: tokenProgramId, isSigner: false, isWritable: false },
-  ];
+class TokenSwapLegacy {
+  // @ts-ignore: TS7019
+  static createInitSwapInstruction(...args): TransactionInstruction {
+    throw new Error('createInitSwapInstruction not available for legacy swap program');
+  }
 
-  let data = Buffer.alloc(1024);
-  if (isLatest) {
-    const fields = [
+  static swapInstruction(
+    tokenSwap: PublicKey,
+    authority: PublicKey,
+    _userTransferAuthority: PublicKey,
+    userSource: PublicKey,
+    poolSource: PublicKey,
+    poolDestination: PublicKey,
+    userDestination: PublicKey,
+    poolMint: PublicKey,
+    feeAccount: PublicKey,
+    _hostFeeAccount: PublicKey | null,
+    swapProgramId: PublicKey,
+    tokenProgramId: PublicKey,
+    amountIn: number | Numberu64,
+    minimumAmountOut: number | Numberu64,
+  ): TransactionInstruction {
+    const dataLayout = BufferLayout.struct([
       BufferLayout.u8("instruction"),
-      BufferLayout.u8("nonce"),
-      BufferLayout.nu64("tradeFeeNumerator"),
-      BufferLayout.nu64("tradeFeeDenominator"),
-      BufferLayout.nu64("ownerTradeFeeNumerator"),
-      BufferLayout.nu64("ownerTradeFeeDenominator"),
-      BufferLayout.nu64("ownerWithdrawFeeNumerator"),
-      BufferLayout.nu64("ownerWithdrawFeeDenominator"),
-      BufferLayout.nu64("hostFeeNumerator"),
-      BufferLayout.nu64("hostFeeDenominator"),
-      BufferLayout.u8("curveType"),
-    ];
+      uint64("amountIn"),
+      uint64("minimumAmountOut"),
+    ]);
 
-    if (config.curveType === CurveType.ConstantProductWithOffset) {
-      fields.push(BufferLayout.nu64("token_b_offset"));
-      fields.push(BufferLayout.blob(24, "padding"));
-    } else if (config.curveType === CurveType.ConstantPrice) {
-      fields.push(BufferLayout.nu64("token_b_price"));
-      fields.push(BufferLayout.blob(24, "padding"));
-    } else {
-      fields.push(BufferLayout.blob(32, "padding"));
+    const keys = [
+          { pubkey: tokenSwap, isSigner: false, isWritable: false },
+          { pubkey: authority, isSigner: false, isWritable: false },
+          { pubkey: userSource, isSigner: false, isWritable: true },
+          { pubkey: poolSource, isSigner: false, isWritable: true },
+          { pubkey: poolDestination, isSigner: false, isWritable: true },
+          { pubkey: userDestination, isSigner: false, isWritable: true },
+          { pubkey: poolMint, isSigner: false, isWritable: true },
+          { pubkey: feeAccount, isSigner: false, isWritable: true },
+          { pubkey: tokenProgramId, isSigner: false, isWritable: false },
+        ];
+
+    // optional depending on the build of token-swap program
+    if (programOwner) {
```

Comment:
I think that is the last bit i need to address and i should be good to go for functional testing